fix(linter): false positive in `eslint/no-redeclare`

shulaoda opened the pull request:
closes #10396 

- Fix the false positive in `typescript`
- Align with `eslint` -- set the default value of `builtinGlobals` to `true`

camchenry commented on crates/oxc_linter/src/rules/eslint/no_redeclare.rs:
```diff
@@ -122,6 +146,8 @@ fn test() {
         ("var globalThis = foo", Some(serde_json::json!([{ "builtinGlobals": false }]))),
         ("var globalThis = foo", Some(serde_json::json!([{ "builtinGlobals": false }]))),
+        // Issue: <https://github.com/oxc-project/oxc/issues/10396>
+        ("export function get(): void; export function get() { }", None),
```
should we include a test for a non-export? like:

```
function foo(arg: string): void; function foo(arg: number): any {}
```

shulaoda replied:
Yes, added it.

shulaoda committed:
fix(linter): false positive in `eslint/no-redeclare` (#10402)

closes #10396

- Fix the false positive in `typescript`
- Align with `eslint` -- set the default value of `builtinGlobals` to `true`
